css/src/components/pageHeader.css

```diff
@@ -30,12 +30,21 @@
   flex-shrink: 0;
 }
 
+.PageHeader-titleWrapper .Button {
+  margin-right: var(--spacing-l);
+  flex-shrink: 0;
+}
+
```

Comment:
`backButton?: React.ReactNode;` backButton has type reactNode so it can ideally accepts any react element not necessarily a Button.  

Also there is a use-case to wrap the backButton inside tooltip. So please consider this case as well.